core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -466,14 +466,14 @@ private boolean expandSelectItem(
     return false;
   }
 
-  private static SqlNode expandExprFromJoin(SqlJoin join, SqlIdentifier identifier,
+  private SqlNode expandExprFromJoin(SqlJoin join, SqlIdentifier identifier,
       @Nullable SelectScope scope) {
-    if (join.getConditionType() != JoinConditionType.USING) {
+    if (join.getConditionType() != JoinConditionType.USING && !join.isNatural()) {
       return identifier;
     }
 
     for (String name
-        : SqlIdentifier.simpleNames((SqlNodeList) getCondition(join))) {
+        : requireNonNull(usingNames(join), () -> "usingNames for " + join)) {
```

Comment:
er, can this be extracted into a variable? it would be more debug friendly

Reply:
Fixed